polyfill/test/Date/prototype/getFields/order-of-operations.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2020 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.date.prototype.getfields
```

Comment:
The operations don't have to delegate back to the Temporal.Date.  They could go straight to the calendar methods:

```javascript
Temporal.Date.prototype.getFields = function() {
  return this.calendar.getFields(this);
}
Temporal.Calendar.prototype.getFields = function(input) {
  return {
    year: this.year(input),
    month: this.month(input),
    day: this.day(input)
  };
};
```

Going back to Temporal.Date would be more like:

```javascript
Temporal.Date.prototype.getFields = function() {
  return this.calendar.getFields(this);
}
Temporal.Calendar.prototype.getFields = function(input) {
  return {
    year: input.year,
    month: input.month,
    day: input.day
  };
};
```

Another option would be to make getFields return a string list:

```javascript
Temporal.Date.prototype.getFields = function() {
  let result = {};
  for (let field of this.calendar.getFields()) {
    result[field] = this[field];  // or this.calendar[field](this);
  }
  return result;
}
Temporal.Calendar.prototype.getFields = function(input) {
  return ["year", "month", "day"];
};
```

Reply:
After working a bit more on the calendar, I think the last option (`Calendar.getFields` returning a list of property keys) is the best one, since it can be used for other things in the calendar implementation as well.

However, I believe that's orthogonal to this pull request; calendar or no calendar, the spec would still require the getters to be called in a particular order.

Reply:
The question is whether the methods get called on the Temporal.Date.prototype, or whether they get called on the Temporal.Calendar.prototype directly.  I'm not sure which is better.

Can you file a follow-up issue?

Reply:
→ #528